### PR TITLE
Add support for PHP 8.1.0

### PR DESCRIPTION
**Pull request by mkoprek**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | no

### Description
Add support for PHP 8.1.0 for laminas/laminas-diagnostics

Drop support for PHP 7.3

Cause there is no any tag yet in **laminas/laminas-code** I used a commit hash

Fixes #30

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+.idea/
```
Comment (samsonasik) [inline .gitignore]: please revert this for concistency, you can add yours in your own system global gitignore

File: src/Check/Memcached.php [hunk 2]
```diff
@@ -72,7 +72,7 @@ public function check()
                 || false === $stats[$authority]
             ) {
                 // Attempt a connection to make sure that the server is really down
-                if (! @$memcached->getLastDisconnectedServer($this->host, $this->port)) {
+                if ($memcached->getLastErrorCode() !== 0) {
```
Comment (sasezaki) [inline src/Check/Memcached.php]: Why changed ?

File: .gitignore (hunk 1 above)
Comment (mkoprek) [reply .gitignore]: Removed

File: src/Check/Memcached.php (hunk 2 above)
Comment (mkoprek) [reply src/Check/Memcached.php]: @sasezaki There was a error that getLastDisconnectedServer() need to have exact 0 argument, but has 2

File: composer.json [hunk 3]
```diff
@@ -27,13 +27,14 @@
...
     "require": {
-        "php": "^7.3 || ~8.0.0",
+        "php": "^7.4 || ~8.0.0 || ~8.1.0",
         "laminas/laminas-zendframework-bridge": "^1.0"
     },
     "require-dev": {
         "doctrine/migrations": "^2.0 || ^3.0",
         "enlightn/security-checker": "^1.2",
         "guzzlehttp/guzzle": "^5.3.3 || ^6.3.3",
+        "laminas/laminas-code": "4.5.x-dev#fbdaecda0c91ea0c7aa81c89eda16b2f345d0657 as 4.5.0",
```
Comment (Ocramius) [inline composer.json]: This should be dropped before merging

File: src/Check/Memcached.php (hunk 2 above)
Comment (Ocramius) [reply src/Check/Memcached.php]: Hmm, this code path doesn't seem to be tested at all: we can't change this without an integration test :thinking:

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: composer.json (hunk 3 above)
Comment (mkoprek) [reply composer.json]: @Ocramius You mean whole line, or just commit sha? The problem is that laminas-code 4.4.* is not supporting PHP 8.1 and 4.5 tag is not pushed yet.

File: src/Check/Memcached.php (hunk 2 above)
Comment (mkoprek) [reply src/Check/Memcached.php]: @Ocramius Ok, so I should go back to previous version and do some if for PHP version?

File: composer.json (hunk 3 above)
Comment (Ocramius) [reply composer.json]: Correct - whole line should be removed. Well aware that `laminas-code` does not support `php: 8.1` yet.

File: src/Check/Memcached.php (hunk 2 above)
Comment (mkoprek) [reply src/Check/Memcached.php]: @Ocramius $memcached->getLastDisconnectedServer return array if there was any error.I changed how it should be handled in here and removed my proposition with getLastErrorCode().

**Comment by mkoprek** [comment]
I removed locking laminas/laminas-code in composer.json
Also I changed if with $memcached->getLastDisconnectedServer(). 

Now all should be good, but there will be problem with composer install until laminas/laminas-code 4.5.0 will be released.

File: composer.json (hunk 3 above)
Comment (mkoprek) [reply composer.json]: Done

**Comment by mkoprek** [comment]
@Ocramius I changes things you mentioned, could you please review it?

**Review by Ocramius — APPROVED** [review]
LGTM :+1: 

Thanks @mkoprek!

**Pull request merged**